### ISSUE TITLE
no-console doesn't work to disable all console statements without an empty "allow" parameter on it.

### ISSUE THREAD

**Issue by chrispinzaru** [eslint/eslint#14008]
https://github.com/eslint/eslint/blob/75fea9bcdd3dde5a07e0089d9011a4df518cdbe3/lib/rules/no-console.js#L75

**Expected to work as follows:**
`no-console: "error"`

**Actually works only when done as follows:**
`"no-console": [
          "error",
          {
            "allow": [""] // ¯\_(ツ)_/¯
          }
 ],`

**Comment by mdjermanovic** [comment]
Hi @chrispinzaru, can you please fill out our [bug report template](https://github.com/eslint/eslint/blob/master/.github/ISSUE_TEMPLATE/BUG_REPORT.md)? We can't know what the issue is without an example where it is observable.

**Comment by DanBoSlice** [comment]
Hi, not original reporter but have the same issue:

**Tell us about your environment**



* Node version: v12.18.0
* npm version: v7.5.4
* Local ESLint version: v7.19.0 (Currently used)
* Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>


```js
{
  "root": true,
  "ignorePatterns": [
    "projects/**/*",
    "src/app/api/**/*"
  ],
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "parserOptions": {
        "project": [
          "tsconfig.json",
          "e2e/tsconfig.json"
        ],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/ng-cli-compat",
        "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "zeph",
            "style": "kebab-case"
          }
        ],
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "zeph",
            "style": "camelCase"
          }
        ],
        "@typescript-eslint/member-ordering": [
          "error",
          {
            "default": [
              "public-static-method",
              "public-constructor"
            ]
          }
        ],
        "@typescript-eslint/no-unused-vars-experimental": "error",
        "no-restricted-imports": [
          "error",
          {
            "patterns": [
              "rxjs/internal/*"
            ]
          }
        ],
        "no-console": "error"
      }
    },
    {
      "files": [
        "*.html"
      ],
      "extends": [
        "plugin:@angular-eslint/template/recommended"
      ],
      "rules": {}
    }
  ]
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**


```js
import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AppComponent implements OnInit {

  ngOnInit(): void {
    console.log('something');
  }

}
```


```bash
node_modules/eslint/bin/eslint.js .
```

**What did you expect to happen?**

`console.log` use should be reported as error.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

All files pass linting.

**Steps to reproduce this issue:**



1. Set "no-console" rule to `"error"`
1. Add console.log('something') in the application
1. Run ESLint. No error is reported.
1. Change "no-console" rule to `["error", {"allow": [""]}]`
1. Re-run ESLint
1. Console statement is now marked as error

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by mdjermanovic** [comment]
@DanBoSlice thanks for the details!

The rule itself seems to be working fine: [Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWNvbnNvbGU6IGVycm9yICovXG5cbmNvbnNvbGUubG9nKCdzb21ldGhpbmcnKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6NSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

The behavior you're seeing is due to how overriding rules' configuration works.

In your example, one of the configuration you're extending from, `plugin:@angular-eslint/ng-cli-compat`, enables `no-console` rule and configures it to allow `.log` and other methods [here](https://github.com/angular-eslint/angular-eslint/blob/94ca59e022ac3977c91720bfbed9b7bd4b31dc51/packages/eslint-plugin/src/configs/ng-cli-compat.json#L129).

```
"no-console": [
      "error",
      {
        "allow": [
          "log",
          "warn",
          "dir",
          "timeLog",
          "assert",
          "clear",
          "count",
          "countReset",
          "group",
          "groupEnd",
          "table",
          "dirxml",
          "error",
          "groupCollapsed",
          "Console",
          "profile",
          "profileEnd",
          "timeStamp",
          "context"
        ]
      }
    ]
```

If you reconfigure the rule in your main config file (derived config) by specifying _only_ severity, ESLint will keep the options from the extended configuration (base config). This behavior is explained in [Extending Configuration Files](https://eslint.org/docs/user-guide/configuring/configuration-files#extending-configuration-files):

> change an inherited rule's severity without changing its options:
> * Base config: "eqeqeq": ["error", "allow-null"]
> * Derived config: "eqeqeq": "warn"
> * Resulting actual config: "eqeqeq": ["warn", "allow-null"]

So, in your example, `"no-console": "error"` has no effect as it overrides just severity but not the options. You can double-check the resulting configuration for a file with the [--print-config](https://eslint.org/docs/user-guide/command-line-interface#-print-config) CLI option.

The solution is to specify options in the main config. `"no-console": ["error", {"allow": [""]}]` or just `"no-console": ["error", {}]` should do the work. That would entirely override the configuration for `no-console` from the extended config.

**Comment by DanBoSlice** [comment]
@mdjermanovic That makes sense! Thanks for the answer :)
